azure/functions/decorators/function_app.py

```diff
@@ -3297,8 +3287,27 @@ def get_functions(self) -> List[Function]:
                 '-bindings-http-webhook-trigger?tabs=in-process'
                 '%2Cfunctionsv2&pivots=programming-language-python#http-auth')
 
+        FunctionRegister.validate_functions(self, functions=functions)
+
         return functions
 
+    def validate_functions(self, functions: List[Function]) -> bool:
```

Comment:
Should we rename this to vaidation_function_names since we are only checking for function names?
Also it wouldn't get mixed up with the other _validation_function  method